sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/Transforms.java

```diff
@@ -381,21 +383,39 @@ static PathItem toPathItem(Path path) {
             parseDateOrNull(path.getLastModified()), path.getContentLength() == null ? 0 : path.getContentLength(),
             path.getGroup(), path.isDirectory() != null && path.isDirectory(), path.getName(), path.getOwner(),
             path.getPermissions(),
-            path.getCreationTime() == null ? null : fromWindowsFileTimeOrNull(Long.parseLong(path.getCreationTime())),
-            path.getExpiryTime() == null ? null : fromWindowsFileTimeOrNull(Long.parseLong(path.getExpiryTime())));
+            parseWindowsFileTimeOrDateString(path.getCreationTime()),
+            parseWindowsFileTimeOrDateString(path.getExpiryTime()));
 
         return AccessorUtility.getPathItemAccessor().setPathItemProperties(pathItem, path.getEncryptionScope(), path.getEncryptionContext());
     }
 
+    static OffsetDateTime parseWindowsFileTimeOrDateString(String date) {
```

Comment:
Something I missed, we should early out on `null` instead of wasting cycles trying to parse it as `long`.